### PR TITLE
Adds Hooks for publishing to crates.io

### PR DESCRIPTION
**Pull request by AshantiMutinta**
## PR summary

(from dave)
ended up being a bit of a rabbit hole..

- some outstanding crates/* path issues
- needed newer holonix to match new deps coming in
- many crates needed more things in their Cargo.toml to stop cargo complaining
- need to add `version =` to all local crates to make them compatible with crates.io
- move sim1h in here as it is a pain to sync with all these now-published crates as a separate repo
- updated the dynamodb nix as per upstream
- adds hooks (i've been testing them manually so far) for versioning and publishing

## testing/benchmarking notes

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

**big followup issue**

the problem is that `dev-dependencies` are considered as a circular dependency when publishing!

this is an upstream bug in cargo because dev dependencies are never published

the "state of the art" is to write a script that rewrites cargo.toml files inline, but i haven't done that here

## changelog

Please check one of the following, relating to the [CHANGELOG-UNRELEASED.md](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md)

- [ ] this is a code change that effects some consumer (e.g. zome developers) of holochain core so it is added to the CHANGELOG-UNRELEASED.md (linked above), with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [x] this is not a code change, or doesn't effect anyone outside holochain core development

### CODE REVIEW

**Comment by thedavidmeister** [comment]
```
Running /tmp/holochaintarget/debug/deps/entry-5bc544db283c7cd1
```

looks suss, missing /

File: crates/common/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "holochain_common"
+description = "holochain core commons"
+license = "Apache-2.0"
```
Comment (zippy) [inline crates/common/Cargo.toml]: Lets change this license to GPLv3 to match other core license please.

File: crates/conductor_api/Cargo.toml [hunk 2]
```diff
@@ -1,14 +1,18 @@
 [package]
 name = "holochain_conductor_lib_api"
+description = "holochain conductor lib api"
+license = "Apache-2.0"
```
Comment (zippy) [inline crates/conductor_api/Cargo.toml]: same: license to gplv3

File: crates/conductor_lib/Cargo.toml [hunk 3]
```diff
@@ -1,23 +1,27 @@
 [package]
 name = "holochain_conductor_lib"
+description = "holochain conductor library"
+license = "Apache-2.0"
```
Comment (zippy) [inline crates/conductor_lib/Cargo.toml]: ditto

File: crates/core/Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,10 @@
 [package]
 name = "holochain_core"
 version = "0.0.34-alpha1"
+description = "holochain core"
+license = "Apache-2.0"
```
Comment (zippy) [inline crates/core/Cargo.toml]: ditto

File: crates/dpki/Cargo.toml [hunk 5]
```diff
@@ -1,13 +1,17 @@
 [package]
 name = "holochain_dpki"
 version = "0.0.34-alpha1"
+description = "holochain dpki"
+license = "Apache-2.0"
```
Comment (zippy) [inline crates/dpki/Cargo.toml]: ditto

File: crates/hdk/Cargo.toml [hunk 6]
```diff
@@ -1,6 +1,10 @@
 [package]
 name = "hdk"
 version = "0.0.34-alpha1"
+description = "holochain hdk"
+license = "Apache-2.0"
```
Comment (zippy) [inline crates/hdk/Cargo.toml]: ditto

File: crates/hdk_v2/Cargo.toml [hunk 7]
```diff
@@ -1,6 +1,10 @@
 [package]
-name = "hdk-proc-macros"
+name = "hdk_proc_macros"
 version = "0.0.34-alpha1"
+description = "holochain hdk v2 (with proc macros)"
+license = "Apache-2.0"
```
Comment (zippy) [inline crates/hdk_v2/Cargo.toml]: ditto

File: crates/sim1h/Cargo.toml [hunk 8]
```diff
@@ -0,0 +1,32 @@
+[package]
+name = "sim1h"
+version = "0.0.34-alpha1"
+authors = ["Holochain Core Dev Team <devcore@holochain.org>"]
+edition = "2018"
+description = "A simulation of lib3h"
+keywords = ["holochain", "holo", "p2p", "network", "simulation"]
+categories = ["network-programming"]
+license = "Apache-2.0"
```
Comment (zippy) [inline crates/sim1h/Cargo.toml]: this one's ok to be apache-2

**Pull request merged**